wordpress-mobile/WordPress-iOS
# Blogging reminders: Unschedule all on logout

This PR unschedules and removes scheduled blogging reminders in a couple of situations:

- If you log out of the default wpcom account, all reminders for all blogs belonging to that account will be removed.
- If you delete a self hosted site from the app, the reminders for that site will be removed.

**To test**

* Build and run, and ensure tests pass
* Log into a wpcom site and add reminders to a site or two. Check that removing reminders works too.
* Add a self hosted site and add some reminders to it
* Log out of the wpcom account and ensure that the self hosted site still has reminders
* Log back into the wpcom account and ensure the reminders didn't persist and were removed

## Regression Notes
1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
